android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -627,7 +675,19 @@ private void isEnabled(Result poResult) {
     /// Method to connect/disconnect wifi service
     private void setEnabled(MethodCall poCall, Result poResult) {
         Boolean enabled = poCall.argument("state");
-        moWiFi.setWifiEnabled(enabled);
+        Boolean shouldOpenSettings = poCall.argument("shouldOpenSettings");
+
+        // Enable or Disable WiFi programmatically
+        if (!shouldOpenSettings) {
+            moWiFi.setWifiEnabled(enabled);
+        }  
+            // Open WiFI settings
+            else {
+                Intent wifiIntent = new Intent(Settings.ACTION_WIFI_SETTINGS);
+                wifiIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                this.moContext.startActivity(wifiIntent);
+            }
+
```

Comment:
- `shouldOpenSettings` should only be used with >= 29.
- also add "null check" for `shouldOpenSettings`

Reply:
@daadu All done, could you review again the changes?